lib/actions/AddLinkToProjectResponse.js

```diff
@@ -1,8 +1,16 @@
 class AddLinkToProjectResponse {
-    constructor(label, link){
-      this.label = label;
-      this.link = link;
+  constructor(label, link){
+    this.label = label;
+    this.link = link;
+  }
+
+  async sendToBot(bot) {
+    try {
+      return bot.reply('Link stored.')
+    } catch {
+      return bot.reply('Failure to store link.')
```

Comment:
This is going to be really hard to debug. What should we do that?

Some things that come to mind, some of which we haven't solved or planned for:
* Include the problem in the reply. Users won't really like that much, but maybe that's ok for our audience.
* Report these to an error reporting "place". Sparkbox relies on rollbar.com for this under normal circumstances.

Reply:
when you say 'report the problem' do you mean that bot.reply will actually spit out the kind of error the catch would be throwing?

Reply:
First, `catch` is not throwing. Something else is `throw`ing. Catch is... `catch`ing 😂.

As for the rest...

If you have a problem `reply`ing with "Link stored.", how likely is it that the "Failure" reply works? Probably not very likely. If something bad happens in your app, don't you want to fix it? How are you going to fix the exception if you don't know what it is. You _could_ `reply` with the exception information. That won't help you if the exception is **caused** by the first `reply` call. Instead, we try to send errors to a place we can record and triage those errors.

Reply:
trueeeeeee. catch doesnt throw. LOL. 
BUT! isnt that where we want to `throw` an error if `reply` fails to work? 
like this -- >
``` 
catch (err) {
  return bot.reply('Failed to store');
  throw new  Error(err);
 }

Reply:
`return` will not occur after a throw. Throw means "I can't handle what's going on, I quit"

It is fine to catch things you can handle. In this case, if you want to catch the inability to "store" your link, great! Do that and `reply` with a message to the user about your issues *AND LOG IT IN A SYSTEM* so you can fix the problem.

If `reply` fails, you should not expect the next `reply` to work. Send an urgent alert to your team to diagnose what's going on.

Reply:
thanks @cromwellryan